error: cast [...] increases required alignment of target type on riscv64

gpui-0.2.46 and gpui-0.2.47 fail to compile on riscv64. I get the following error:

```
In file included from /usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.h:29,
                 from /usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.cpp:22:
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h: In instantiation of 'std::__cxx11::basic_string<_CharT> pol::convert(string_const_iterator<source_char>, string_const_iterator<source_char>, iconv_t) [with target_char = char16_t; source_char = char; string_const_iterator<source_char> = std::__cxx11::basic_string<char>::const_iterator; iconv_t = void*]':
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:179:45:   required from 'std::__cxx11::basic_string<_CharT> pol::convert(const std::__cxx11::basic_string<source_char, std::char_traits<_CharT>, std::allocator<_T2> >&, iconv_t) [with target_char = char16_t; source_char = char; iconv_t = void*]'
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.cpp:76:68:   required from here
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:163:23: error: cast from 'std::array<char, 512>::pointer' {aka 'char*'} to 'char16_t*' increases required alignment of target type [-Werror=cast-align]
  163 |         result.append(reinterpret_cast<target_char *>(temp->data()),
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:164:23: error: cast from 'char*' to 'char16_t*' increases required alignment of target type [-Werror=cast-align]
  164 |                       reinterpret_cast<target_char *>(outbuf));
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/plugins/pol/CMakeFiles/pol-plugin.dir/build.make:76: src/plugins/pol/CMakeFiles/pol-plugin.dir/binary.cpp.o] Error 1
```

--- a/src/plugins/pol/encoding.h
+++ b/src/plugins/pol/encoding.h
@@ -149,19 +149,18 @@ inline std::basic_string<target_char> convert(string_const_iterator<source_char>
     char *inbuf = reinterpret_cast<char *>(const_cast<source_char *>(&*begin));
     size_t inbytesLeft = std::distance(begin, end) * sizeof(source_char);
 
-    auto temp = std::make_unique<std::array<char, 512>>();
-    char *outbuf = temp->data();
+    auto temp = std::make_unique<std::array<target_char, 512>>();
+    target_char *outbuf = temp->data();
     size_t outbytesLeft = temp->size();
 
     while (inbytesLeft > 0) {
-        auto ret = iconv(conv, &inbuf, &inbytesLeft, &outbuf, &outbytesLeft);
+        auto ret = iconv(conv, &inbuf, &inbytesLeft, reinterpret_cast<char**>(&outbuf), &outbytesLeft);
         if (ret == ICONV_ERROR_CODE && errno != E2BIG) {
             throw std::runtime_error("LINE: " + std::to_string(__LINE__) + ", FILE: " + __FILE__
                                      + ", Encountered corrupted unicode string.");
         }
 
-        result.append(reinterpret_cast<target_char *>(temp->data()),
-                      reinterpret_cast<target_char *>(outbuf));
+        result.append(temp->data(), outbuf);
         outbuf = temp->data();
         outbytesLeft = temp->size();
     }

